src/utils/networkHealth.ts

```diff
@@ -0,0 +1,89 @@
+import { NetworkHealth } from 'interfaces/networkHealth.interface'
+import { BlockTimeHealth } from '../interfaces/blockTimeHealth.interface'
+import { getLatestBlocks } from '../repositories/block.repository'
+import { logger } from './logger'
+import { networkHealthConfig } from './networkHealthConfig'
+import { networkHealthNotification } from './notifications'
+
+export async function checkNetworkHealth() {
+  const blockTimeHealth = await calculateBlockTimeHealth()
+  if (!blockTimeHealth) return
```

Comment:
question(blocking): When could this happen? Will there be no check and notification at all then?

Reply:
1. This can happen when there are not enough blocks to analyze(nothing happens => no notification, this case is meant for when the block fetching starts with empty db) 
2. The calculations fails due to an error and null is returned (error is logged in the catch within `calculateBlockTimeHealth`)